### ISSUE TITLE
Implement UserTag in JavaScript

### ISSUE THREAD

**Issue by Fox32** [dart-lang/sdk#25829]
Would it be possible to implement the [```UserTag```](https://api.dartlang.org/1.14.2/dart-developer/UserTag-class.html) API from ```dart:developer``` in JavaScript? Right now it is only a dummy implementation. But one could implement a simple subtition for dart2js with [```window.performance.mark()```](https://developer.mozilla.org/en-US/docs/Web/API/Performance/mark) and [```window.performance.measure()```](https://developer.mozilla.org/en-US/docs/Web/API/Performance/measure).
I'm not sure about the performance implications of this change. But with the [upcomming support of the Chrome DevTools for ```performance.measure()``` in the timeline](https://plus.google.com/+IlyaGrigorik/posts/27ATwubWn78) this would be a cool thing. This is especially useful for bigger, performance sensitive applications running in a browser.

**Comment by hterkelsen** [comment]
Interesting idea. If I understand this correctly, we would want something like this:
* `UserTag#makeCurrent`: call `window.performance.mark('$currentTag')`, `window.performance.measure('$previousTag', '$previousTag', '$currentTag')`

**Comment by Fox32** [comment]
That would be an option, on the other hand I'm not sure how useful this feature request would be at all. The semantic would be quite different between Dart and JS.

**Comment by hterkelsen** [comment]
Yeah, it looks like the `Timeline` API maps more cleanly to the javascript `Performance` API. Can't promise we'll get to this anytime soon, though.
